Proposed product descriptions/Execution Engine - Components Consolidation.md

```diff
@@ -0,0 +1,51 @@
+## Execution Engine Update - Interface Consolidation
+
+### Summary
+KBase currently lacks a centralized execution engine interface, which has generated a significant amount of unnecessary complexity, system instability issues, and lingering problems with job and narrative sharing and security.  There are many possible Execution Engine related projects (HPC execution, resource limits per user, BYOC), but all require (or will be significantly simplified) by consolidation of the existing interfaces into a single execution engine service first.  Proof of delivery will be evident in a new Jobs Dashboard page or widget.
+
+### Story / Description
+The current Job related services that are accessed directly through various components are: NarrativeJobService (NJS), NarrativeJobServiceWrapper (NJS_wrapper), NarrativeJobProxy (NJP), User and Job State (UJS), AWE Execution Engine.  Instead, the Narrative and all Job checking operations will be migrated to a single, well-tested service.  The service will handle submission of jobs, listing of current and past jobs, sharing of jobs automatically (as jobs should be attached to a workspace/narrative), fetching job state, fetching job input/output, fetching job logs, and to the extent possible by the job dispatch system (AWE or its replacement) setting some limits on job execution time or jobs per user.  The implementation team will determine the extent to which UJS/NJS/NJP/NJS Wrapper can be reused, salvaged, or completely replaced.
```

Comment:
My first instinct is to just not share or copy job information on a narrative copy.  On a copy, we should probably clear out that jobs list entirely.  Especially if the job is actively running, and we want it to appear as the job is running in the copy, it means the results would also need to be copied after the job finishes, which seems like odd behavior.  Better I think to just give a warning/error message in that case.

The only problem with not copying job info is that the console log is attached to jobs right now (which could be changed), so completed logs wouldn't be visible in a narrative that is copied.  But console logs should not be used to store information that needs to persist or be operated on anyway- that's what output data and the new Report type is for.  Console logs should probably only be used for checking progress or debugging issues, so I think that it's ok if copied logs are no longer visible.